### DISCUSSION TITLE
Solving system of coupled PDEs and an ODE

### DISCUSSION THREAD

**Discussion by Rahul-Patel25** [zwicker-group/py-pde#358] [Q&A] [answered]
I am trying to model a tubular reactor where the governing equations are a system of PDE and an ODE. The ODE is not time-varying and is only dependent on a spatial variable. 
I have been able to to solve multiple coupled PDEs with their Boundary conditions and wanted to check if it was possible to solve system of ODE and PDEs by this package.

For reference, I am trying to mimic the example given here : http://websites.umich.edu/~elements/web_mod/radialeffects/unsteady/index1.htm#model 

![image](https://user-images.githubusercontent.com/97452995/214269013-51a4e487-2e49-4936-93e2-6d803b039a1c.png)
![image](https://user-images.githubusercontent.com/97452995/214269124-34634dbe-43fe-4874-b03d-1659614570d2.png)
![image](https://user-images.githubusercontent.com/97452995/214269401-ee794828-dd70-4630-8ce3-b02e82849b2b.png)

Is the py-pde package capable of solving such systems or are there any easy workarounds to address this?

**Comment by david-zwicker** [comment, marked as answer]
I'm afraid I don't understand the problem. As far as I can see, `C_i` and `T` are both fields depending on `(r, z, t)` implying that the last equation also depends on these three variables, so all equations are PDEs. It seems as if I'm missing something.

**Reply by Rahul-Patel25** [reply]
Thank you for the response. I am unsure about how to incorporate the last equation (Using PDE class) since it doesn't have a time-derivative term (unlike the first two equations) and wanted to know how to do it.

For a sample/representative code, I have incorporated the first two equations (For i=A,B,C) as below.
![image](https://user-images.githubusercontent.com/97452995/214475848-b8304cb7-8400-46bd-8b8a-aa8e7e03aff1.png)

**Reply by david-zwicker** [reply]
I honestly do not understand the last equation. What would be appropriate boundary conditions? What happens to the `r`-dependence of the `T`-field?

While `py-pde` always expects PDEs to be formulated with an explicit time dependence, auxiliary ODEs/PDEs can sometimes be included by introducing a fictitious time-dependence, i.e., some dynamics that relax toward the solution of the actual equation.
